From 9dad91a8465ba685f7143e6dd8b79fad262bb27b Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Wed, 8 Jan 2020 14:26:10 +0200
Subject: [PATCH 1412/1921] dts: marvell: add xor DMA copy offload AP support

Add node support for AP xor offload copy driver.
Replace the two xor engines used on OCP platform with
xor DMA nodes.

Change-Id: I4d134a3b45c87251c891ace6848a322bc8e95333
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/21232
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49164
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap80x.dtsi | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
index 7b0c008a1382..3baf814487f4 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
@@ -205,6 +205,26 @@
 				dma-coherent;
 			};
 
+			dma_xor0: dma-xor@400000 {
+				compatible = "marvell,mv-xor-v2-copy-offload";
+				reg = <0x400000 0x1000>,
+				      <0x410000 0x1000>;
+				msi-parent = <&gic_v2m0>;
+				clocks = <&ap_clk 3>;
+				dma-coherent;
+				status = "disabled";
+			};
+
+			dma_xor1: dma-xor@420000 {
+				compatible = "marvell,mv-xor-v2-copy-offload";
+				reg = <0x420000 0x1000>,
+				      <0x430000 0x1000>;
+				msi-parent = <&gic_v2m0>;
+				clocks = <&ap_clk 3>;
+				dma-coherent;
+				status = "disabled";
+			};
+
 			uio_xor0 {
 				compatible = "marvell,uio-xor-v2";
 				xor_access = <&xor0>;
-- 
2.31.1

